### PR TITLE
⏲ Added Time Range Component

### LINKED ISSUE #5: Create time range selector
**Summary**
Create panel and range selector.
image

**Acceptance Criteria / User Stories**

- As a user, I should be able to
    - [ ] click and drag the handles to set a time range (2000 to present for now).

### PR DESCRIPTION
**Pull request by abetco**
### Changes
Created the time range component.

As of now works for year dates, could be adapted to different formats as necessary.

Props (all values are ints) : 
- range: the current range of values, will be updated by the component when the user changes the input
- setRange: setter for range
- minTime: leftmost time on the range
- maxTime: rightmost time on the range
- step: step size between times

Built into the component are aria labels for both handles (min, max, current value), and it is tabbable as well.

Additionally, did some restructuring to create a map page and added the map component as well as the time range slider to the page.

Maybe could use some styling changes later when the sidebar container is integrated.

Addresses #5 

### Screenshots (if UI change)

image

image

### CODE REVIEW

File: client/src/components/TimeRange.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// @flow
+
+import React, { useEffect, useState } from "react";
+import { Range } from "rc-slider";
+import "rc-slider/assets/index.css";
+import "./../styles/TimeRange.css";
+
+const TimeRange = (props: Props) => {
```
Comment (kenetec) [inline client/src/components/TimeRange.js]: Can you define the props for the component?

File: client/src/components/TimeRange.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    }
+    setMarks(newMarks);
+  }, [maxTime, minTime, setMarks, step]);
+
+  const onChange = (newRange) => {
+    setRange(newRange);
+  };
+
+  return (
+    <div className="rangeSlider">
```
Comment (kenetec) [inline client/src/components/TimeRange.js]: Let's add a role prop to the div as well.

**Review by kenetec — APPROVED** [review]
Awesome work! Just a few changes to make before you merge! 🤩

**Review by rxun — COMMENTED** [review]
it looks good (thanks for aria features), but is it keyboard accessible? if it's too tricky to handle, then i think we can ask if siraj can maybe incorporate a more accessible design for the future

**Comment by abetco** [comment]
> it looks good (thanks for aria features), but is it keyboard accessible? if it's too tricky to handle, then i think we can ask if siraj can maybe incorporate a more accessible design for the future

You can tab to each of the handles on the slider, and adjust the value using the arrow keys. Is there anything else in mind that you had for keyboard accessibility?

**Pull request merged**